stl/src/filesys.cpp

```diff
@@ -226,23 +221,27 @@ _FS_DLL unsigned long long __CLRCALL_PURE_OR_CDECL _Hard_links(const wchar_t* _F
     HANDLE _Handle = _FilesysOpenFile(_Fname, FILE_READ_ATTRIBUTES, FILE_FLAG_BACKUP_SEMANTICS);
 
 #ifdef _CRT_APP
+    if (_Handle == INVALID_HANDLE_VALUE) {
+        return static_cast<unsigned long long>(-1);
+    }
+
     FILE_STANDARD_INFO _Info = {0};
-    bool _Ok                 = false;
 
-    if (_Handle != INVALID_HANDLE_VALUE) { // get file info
-        _Ok = GetFileInformationByHandleEx(_Handle, FileStandardInfo, &_Info, sizeof(_Info)) != 0;
-        CloseHandle(_Handle);
-    }
-    return _Ok ? _Info.NumberOfLinks : static_cast<unsigned long long>(-1);
+    // get file info
+    const auto _Ok = GetFileInformationByHandleEx(_Handle, FileStandardInfo, &_Info, sizeof(_Info));
+    CloseHandle(_Handle);
+    return _Ok != 0 ? _Info.NumberOfLinks : static_cast<unsigned long long>(-1);
 #else // _CRT_APP

```

Comment:
These preprocessor guards can be enhanced to `^^^ / vvv` (but possibly in a separate change, see #351)